### PR TITLE
15890-remove confirm name change checkbox for corp correction

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#15890

*Description of changes:*

-  For Staff Corrections of Director data, please remove this line from the screen:
image
-  Applicable to Correction filings only and corporations only.
-  Remove the box all-together

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/PeopleAndRoles/OrgPerson.vue [hunk 1]
```diff
@@ -745,7 +745,7 @@ export default class OrgPerson extends Mixins(CommonMixin, OrgPersonMixin) {
   /** Returns True if person name or org name has changed from its original properties. */
   protected hasNameChanged (orgPerson: OrgPersonIF): boolean {
     // is this a pre-existing person?
-    if (this.isPreExisting && this.isPerson) {
+    if (this.isFirmCorrectionFiling && this.isPreExisting && this.isPerson) {
```
Comment (ketaki-deodhar) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:748]: Before - change first name for CCC. 
image

After - change first name for CCC.
image

Checkbox still displayed for firms.
image

Comment (severinbeauvais) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:748]: Please add another comment explaining that this check is for firm corrections only (does not apply to corps).

In which other filings is this component used and needs to be tested? Conversions maybe?

Comment (severinbeauvais) [reply src/components/common/PeopleAndRoles/OrgPerson.vue:748]: What about the applicant for a limited restoration conversion?

Comment (ketaki-deodhar) [reply src/components/common/PeopleAndRoles/OrgPerson.vue:748]: I will test other types of filings a bit more

File: src/components/common/PeopleAndRoles/OrgPerson.vue [hunk 2]
```diff
@@ -744,8 +744,9 @@ export default class OrgPerson extends Mixins(CommonMixin, OrgPersonMixin) {
 
   /** Returns True if person name or org name has changed from its original properties. */
   protected hasNameChanged (orgPerson: OrgPersonIF): boolean {
+    // This check is only for firm corrections and does not apply to corps corrections
     // is this a pre-existing person?
-    if (this.isPreExisting && this.isPerson) {
+    if (!this.isBenBcCccUlcCorrectionFiling && this.isPreExisting && this.isPerson) {
```
Comment (ketaki-deodhar) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:749]: I think this is the correct condition to check. Tested Conversion, Change, Extension, Conversion filings.

**Review by jonathan-longe — APPROVED** [review]
I'm not familiar with the PeopleAndRoles code, but from what I can see, this looks good.

**Review by JazzarKarim — APPROVED** [review]
Looks good to me! 👍

File: src/components/common/PeopleAndRoles/OrgPerson.vue [hunk 3]
```diff
@@ -636,6 +636,14 @@ export default class OrgPerson extends Mixins(CommonMixin, OrgPersonMixin) {
     return this.wasReplaced(this.currentOrgPerson) ? '' : 'ACTIVE'
   }
 
+  /** Flag to show confirm name change checkbox.
+  This check is only for firm corrections, change, conversion, Corp extension and Corp conversion filings.
+  Does not apply to corps corrections */
+  get showConfirmNameChange () : boolean {
```
Comment (ketaki-deodhar) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:642]: Moved comment

File: src/components/common/PeopleAndRoles/OrgPerson.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/common/PeopleAndRoles/OrgPerson.vue]: Resolving as the logic looks correct now.

File: src/components/common/PeopleAndRoles/OrgPerson.vue (hunk 3 above)
Comment (severinbeauvais) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:642]: I know naming things is hard, but this is only one condition for identifying that the name has changed. (The other condition is on line 765.)

I can't think of a better name, but I think the code would look OK if you made this logic a variable inside hasNameChanged() and then used that variable on line 756.

File: src/components/common/PeopleAndRoles/OrgPerson.vue [hunk 4]
```diff
@@ -744,8 +744,13 @@ export default class OrgPerson extends Mixins(CommonMixin, OrgPersonMixin) {
...
   protected hasNameChanged (orgPerson: OrgPersonIF): boolean {
-    // is this a pre-existing person?
-    if (this.isPreExisting && this.isPerson) {
+    /** This check is only for firm corrections, change, conversion, Corp extension and Corp conversion filings.
+    Does not apply to corps corrections */
+    const showConfirmNameChange = this.isFirmCorrectionFiling || this.isFirmChangeFiling ||
+      this.isFirmConversionFiling || this.isLimitedRestorationExtension || this.isLimitedRestorationToFull
+
+    // check showConfirmNameChange and is this a pre-existing person?
+    if (showConfirmNameChange && this.isPreExisting && this.isPerson) {
```
Comment (ketaki-deodhar) [inline src/components/common/PeopleAndRoles/OrgPerson.vue:753]: @severinbeauvais, something like this? thanks for your earlier comment :)

**Review by severinbeauvais — APPROVED** [review]
LGTM.

By the way, you don't have to re-request a review from me -- I already get an email that notification that you updated something.

**Comment by ketaki-deodhar** [comment]
> LGTM.
> 
> By the way, you don't have to re-request a review from me -- I already get an email that notification that you updated something.

Okay :)

**Comment by ketaki-deodhar** [comment]
@severinbeauvais, thanks for merging.

**Pull request merged**